taichi/codegen/spirv/spirv_codegen.cpp

```diff
@@ -567,7 +567,10 @@ class TaskCodegen : public IRVisitor {
   void visit(ArgLoadStmt *stmt) override {
     const auto arg_id = stmt->arg_id;
     const auto arg_type = ctx_attribs_->args_type()->get_element_type({arg_id});
-    if (arg_type->is<PointerType>()) {
+    if (arg_type->is<PointerType>() ||
+        (arg_type->is<lang::StructType>() && arg_type->as<lang::StructType>()
```

Comment:
I think we shouldn't add NDarray type (complicates IR), but rather rely on annotation information or the context buffer argument type.

Reply:
You mean we should use `KernelContextAttributes::arg_attribs_vec_` or `Callable::parameter_list`? (btw we should unify them)